Realm/RLMArrayLinkView.mm

```diff
@@ -375,7 +375,16 @@ - (RLMResults *)objectsWithPredicate:(NSPredicate *)predicate {
 - (NSUInteger)indexOfObjectWithPredicate:(NSPredicate *)predicate {
     auto query = translateErrors([&] { return _backingList.get_query(); });
     query.and_query(RLMPredicateToQuery(predicate, _objectInfo->rlmObjectSchema, _realm.schema, _realm.group));
+#if REALM_VER_MAJOR >= 2
```

Comment:
The behavior change was made on master but I don't think a core release has been made since it was landed.